consensus/polybft/consensus_runtime.go

```diff
@@ -0,0 +1,851 @@
+package polybft
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"sync"
+	"sync/atomic"
+
+	"github.com/0xPolygon/pbft-consensus"
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/bitmap"
+	bls "github.com/0xPolygon/polygon-edge/consensus/polybft/signer"
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/wallet"
+	"github.com/0xPolygon/polygon-edge/types"
+	hcf "github.com/hashicorp/go-hclog"
+	"github.com/umbracle/ethgo"
+	"github.com/umbracle/ethgo/abi"
+)
+
+const (
+	eventsBufferSize   = 10
+	stateFileName      = "consensusState.db"
+	uptimeLookbackSize = 2 // number of blocks to calculate uptime from the previous epoch
+)
+
+var (
+	// state sync metrics
+	// TODO: Nemanja - accommodate bridge metrics to Edge
+	// totalStateSyncsMeter = metrics.NewRegisteredMeter("consensus/bridge/stateSyncsTotal", nil)
+
+	// errNotAValidator represents "node is not a validator" error message
+	errNotAValidator = errors.New("node is not a validator")
+	// errQuorumNotReached represents "quorum not reached for commitment message" error message
+	errQuorumNotReached = errors.New("quorum not reached for commitment message")
+)
+
+// Transport is an abstraction of network layer
+type Transport interface {
+	Gossip(message interface{})
+}
+
+// epochMetadata is the static info for epoch currently being processed
+type epochMetadata struct {
+	// Number is the number of the epoch
+	Number uint64
+
+	// LastCheckpoint is the last epoch that was checkpointed, for now it is epoch-1.
+	LastCheckpoint uint64
+
+	// CheckpointProposer is the validator that has to send the checkpoint, assume it is static for now.
+	CheckpointProposer pbft.NodeID
+
+	// Blocks is the list of blocks that we have to checkpoint in rootchain
+	Blocks []*types.Block
+
+	// Validators is the set of validators for the epoch
+	Validators AccountSet
+
+	// Commitment built in the current epoch
+	Commitment *Commitment
+}
+
+// runtimeConfig is a struct that holds configuration data for given consensus runtime
+type runtimeConfig struct {
+	PolyBFTConfig  *PolyBFTConfig
+	DataDir        string
+	Transport      Transport
+	Key            *wallet.Key
+	State          *State
+	blockchain     blockchainBackend
+	polybftBackend polybftBackend
+}
+
+// consensusRuntime is a struct that provides consensus runtime features like epoch, state and event management
+type consensusRuntime struct {
+	// config represents wrapper around required parameters which are received from the outside
+	config *runtimeConfig
+
+	// state is reference to the struct which encapsulates bridge events persistence logic
+	state *State
+
+	// eventTracker is a reference to the log event tracker
+	eventTracker *eventTracker
+
+	// epoch is the metadata for the current epoch
+	epoch *epochMetadata
+
+	// lock is a lock to access 'epoch'
+	lock sync.Mutex
```

Comment:
Drop this lock member, and embed the `sync.Mutex` interface directly

Reply:
this is against our approach. @ferranbt  can elaborate on this